api/internal/target/maker_test.go

```diff
@@ -7,6 +7,7 @@ import (
 	"testing"
 
 	pLdr "sigs.k8s.io/kustomize/api/internal/plugins/loader"
+	depproviderapi "sigs.k8s.io/kustomize/api/internal/provider"
 	"sigs.k8s.io/kustomize/api/internal/target"
 	fLdr "sigs.k8s.io/kustomize/api/loader"
 	"sigs.k8s.io/kustomize/api/provider"
```

Comment:
Can we remove all of kustomize usage of the deprecated "sigs.k8s.io/kustomize/api/provider"?

You won't be able to replace calls to `provider` that are external to the `api` module, but I think you can replace those with calls to the `DepProvider` struct's fields, like `resource.Factory` and `ifc.Validator` instead.

Reply:
Replied in the previous comment to understand if we want to deprecate it. Since I'm fairly new to the project, I'm not sure of the users who would be affected by this.